packages/listview/src/components/ColoredStatusColumn/index.js

```diff
@@ -1,30 +1,98 @@
-import React from 'react';
+import React, { useCallback, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
-import useDefaultColors from './hooks/useDefaultColors';
+import InternalOverlay from 'react-rainbow-components/components/InternalOverlay';
+import { RenderIf } from 'react-rainbow-components';
+import { useOutsideClick, useWindowResize } from 'react-rainbow-components/libs/hooks';
+import { HeaderOption } from './option';
 import getBackgroundColor from './helpers/getBackgroundColor';
-import getNormalizedColors from './helpers/getNormalizedColors';
 import getColor from './helpers/getColor';
-import { StyledContainer } from './styled';
+import getNormalizedColors from './helpers/getNormalizedColors';
+import useDefaultColors from './hooks/useDefaultColors';
+import resolvePosition from './helpers/resolvePosition';
+import { StyledContainer, StyledDropdown, StyledIndicator } from './styled';
+import Options from './options';
+import { dropdownHeaderLabel } from './labels';
+
+const ColoredStatusColumn = ({
+    row,
+    value,
+    colors,
+    textTransform,
+    isEditable,
+    onChange,
+    dropdownHeader,
+}) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const containerRef = useRef();
+    const dropdownRef = useRef();
 
-const ColoredStatusColumn = ({ value, colors, textTransform }) => {
     const defaultColors = useDefaultColors();
     const map = colors ? getNormalizedColors(colors) : defaultColors;
     const backgroundColor = getBackgroundColor(map[value.toLowerCase()]);
     const color = getColor(map[value.toLowerCase()]);
+    const ariaHaspopup = isEditable ? 'listbox' : null;
+    const header = dropdownHeader || dropdownHeaderLabel;
```

Comment:
we need to discuss this prop name and then if we pass this as a prop then we don't need the dropdownHeaderLabel